src/Compilers/CSharp/Portable/Symbols/Tuples/TupleErrorFieldSymbol.cs

```diff
@@ -124,7 +124,10 @@ public bool Equals(TupleErrorFieldSymbol other)
                 return true;
             }
 
-            return (object)other != null && _tupleFieldId == other._tupleFieldId && ContainingType == other.ContainingType;
+            return (object)other != null &&
+                _tupleElementIndex == other._tupleElementIndex &&
+                ContainingType == other.ContainingType &&
+                Name == other.Name;
```

Comment:
Currently {index, name} combination is unique to all regular tuple fields by the way of constructing - we do compare names there before creating fields. Individual Name or Index are not unique though.
It is probably the case for error field symbols as well.